pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/store/TableViewLoadDataStoreImpl.java

```diff
@@ -61,30 +70,58 @@ public CompletableFuture<Void> removeAsync(String key) {
 
     @Override
     public Optional<T> get(String key) {
+        validateTableViewStart();
```

Comment:
It seems that if the check `validateTableViewStart` is failed, the UnloadScheduler will also fail to find bundles that are needed to unload, and the loadManager can't assign a broker for the bundle, can we try to re-initialize the tableView in the method `validateTableViewStart`?

Reply:
When ExtensibleLoadManager.start() and playLeader() are complete, then topBundlesLoadStore.validateTableViewStart() should never fail from the leader broker(from UnloadScheduler).

Currently, follower brokers should never access topBundlesLoadDataStore.tableview, and if they do, we should throw an exception in this case instead of re-initializing the tableView(followers don't run UnloadScheduler).

I also moved the UnloadScheduler.start()/close() logic into the playLeader() and playFollower().

Reply:
According to this code block, even if it failed to start the `topBundlesLoadDataStore`,  the `playLeader` still can complete, if the UnloadScheduler tries to find bundles that need to unload, it will encounter IllegalStateException("table view has not been started").

```
    private void playLeader() {
        this.pulsar.getLoadManagerExecutor().execute(() -> {
            serviceUnitStateChannel.scheduleOwnershipMonitor();
            waitForLoadStoreInit();
            try {
                topBundlesLoadDataStore.startTableView();
            } catch (Throwable e) {
                log.error("The new leader:{} failed to start topBundlesLoadDataStore tableview",
                        pulsar.getLookupServiceAddress(), e);
            }
            ...
            log.info("This broker:{} is the leader now.", pulsar.getLookupServiceAddress());
        });
    }

```

Reply:
> ExtensibleLoadManager.start() and playLeader() are complete

I meant this `complete` as everything initialized correctly without an error.

Yes. Currently, if `topBundlesLoadDataStore.startTableView();` fails, the leader broker still proceeds, and the TransferShedder will constantly make UnloadDecision(fail). 

However, lazy-init(tableview.start ) in validateTableViewStart could also fail and lead to the same situation (TransferShedder will constantly make UnloadDecision(fail)). Besides, this lazy-init can lead to unknown behaviors(many consumers on this topBundlesLoad topic) when other brokers access the tableview.

The bigger question is how Pulsar handles system state change failures. In this case, the state change is from playLeader or playFollower.

I can think of the following options:

Option 1: retry-forever
Option 2: fail-fast(shutdown broker)
Option 3: ignore (leader or followers will run in the invalid state, and there will be many error logs and metrics. Eventually, the operator needs to capture this and fix it, probably by broker restart)

AFAIK, upon system state change failures, other systems immediately fail fast with meaningful logs and even with coredump. However, I don't think this `fail-fast` is the practice in the Pulsar project. 

I can update the code for option 1 (retry-forever).

Please let me know if you have more concerns.

Reply:
Obviously, if everything is initialized correctly without an error, it's OK, but you caught exceptions while starting topBundlesLoadDataStore before.

> However, lazy-init(tableview.start ) in validateTableViewStart could also fail and lead to the same situation (TransferShedder will constantly make UnloadDecision(fail))

Yes, the lazy initialization in validateTableViewStart could also fail, but it will not fail forever. If we only caught the exception, it will fail forever.

> I can update the code for option 1 (retry-forever).

I think it's ok, we can merge it first, and we can make more tests in the future.